include/swift/Basic/type_traits.h

```diff
@@ -29,7 +29,7 @@ namespace swift {
 /// is not intended to be specialized.
 template<typename T>
 struct IsTriviallyCopyable {
-#if _LIBCPP_VERSION
+#if _LIBCPP_VERSION || defined(_MSC_VER)
```

Comment:
The value for _MSC_VER will be the same across clang and clang-cl.  What exactly are you trying to protect against?